pymc3/sampling.py

```diff
@@ -1027,8 +1028,11 @@ def stop_tuning(step):
     return step
 
 
-def sample_posterior_predictive(trace, samples=None, model=None, vars=None, size=None,
+def sample_posterior_predictive(trace, samples=None, model=None, vars=None,
+                                varnames=None,
+                                size=None,
                                 random_seed=None, progressbar=True):
+    # type: (MultiTrace, Optional[int], Optional[Model], Optional[List[RV]], Optional[List[str]], Optional[int], Optional[Any], bool) -> Dict[str, np.ndarray]:
```

Comment:
No need to comment this out. Type hints are ok because they're supported by python 3.5. What we don't support is variable annotation.
However, the line is too long. Try splitting it with the `black` package. Don't `black` the entire file, just the `sample_posterior_predictive` function.

Reply:
Actually, this type hint functions despite being commented out. I'm not sure what the alternative to this form of type annotation for pre-3.6 Python.
So can I use this form in 3.5?

``` python
def sample_posterior_predictive(trace, 
                                                    samples: Optional[int]=None, 
                                                    model: Optional[Model]=None,
                                                    vars: Optional[Iterable[Tensor]]=None,
                                                    varnames: Optional[Iterable[str]]=None,
                                                    size: Optional[int]=None,
                                                    random_seed=None,
                                                    progressbar: bool=True) -> Dict[str, np.ndarray]:
```
I'm afraid I don't understand what is and isn't permitted in 3.5, since I went straight from 2.7 to 3.6.
[Nor, obviously, do I understand what the heck Markdown will do with my whitespace!]

Reply:
Type hints are specified in [pep484](https://www.python.org/dev/peps/pep-0484/). How to annotate callables is [here](https://www.python.org/dev/peps/pep-0484/#callable). What you suggest seems to be variable annotation instead

Reply:
PEP484 contains this example, which is of the same form as what I gave in my response:

```
def dilate(v: Iterable[Tuple[T, T]], scale: T) -> Iterable[Tuple[T, T]]:
```

I have *also* type hinted local variables, but I moved all that to comments. So I think the example above should be ok with 3.5, if it supports PEP484.

Reply:
Ah, great!